services/app-web/src/components/SubmissionSummarySection/RateDetailsSummarySection/RateDetailsSummarySection.tsx

```diff
@@ -46,8 +64,86 @@ export const RateDetailsSummarySection = ({
     )
     const isSubmitted = submission.status === 'SUBMITTED'
     const isEditing = !isSubmitted && navigateTo !== undefined
-    //Checks if submission is a previous submission
     const isPreviousSubmission = usePreviousSubmission()
+
+    // Return refreshed package names for state  - used rates across submissions feature
+    // Use package name from api if available, otherwise use packageName coming down from proto as fallback
+    // Display package name appended with text "Draft" when CMS user loads page properly (no errors) but refreshed name is not available
+    const refreshPackagesWithSharedRateCert = (
+        rateInfo: RateInfoType
+    ): SharedRateCertDisplay[] | undefined => {
+        return rateInfo.packagesWithSharedRateCerts?.map(
+            ({ packageId, packageName }) => {
+                const refreshedName =
+                    packageId &&
+                    packageNamesLookup &&
+                    packageNamesLookup[packageId]?.packageName
+                const isDraftText =
+                    isCMSUser && !refreshedName && !indexPackagesError
+                        ? ' (Draft)'
+                        : ''
+                return {
+                    packageId,
+                    packageName:
+                        refreshedName ?? `${packageName}${isDraftText}`,
+                }
+            }
+        )
+    }
+
+    // Request updated rate packages and names for the state  -  used in rates across submissions feature
+    const {
+        error: indexPackagesError,
+        data,
+        loading,
+    } = useIndexHealthPlanPackagesQuery()
+    const refreshedPackagesLookup = data?.indexHealthPlanPackages.edges.reduce(
+        (acc, edge) => {
+            const pkg = edge.node
+            const currentRevisionPackageOrError =
+                getCurrentRevisionFromHealthPlanPackage(pkg)
+            if (currentRevisionPackageOrError instanceof Error) {
+                recordJSException(
+                    `indexHealthPlanPackagesQuery: Error decoding proto. ID: ${pkg.id}`
+                )
+                return acc
+            }
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            const [currentRevision, currentSubmissionData] =
+                currentRevisionPackageOrError
+
+            // Exclude active submission package and contract_only from list.
+            if (
+                currentSubmissionData.submissionType === 'CONTRACT_AND_RATES' &&
+                currentRevision.submitInfo?.updatedAt &&
+                pkg.id
```

Comment:
Currently linked draft submissions will not update the package name when the linked draft submission changes its programs It will update the package name once the linked draft submission is submitted. Is this intended?

Reply:
This is a great catch! Thank you - I cleaned this up by just making that lookup a lot simpler. [f2eda5b](https://github.com/CMSgov/managed-care-review/pull/1349/commits/f2eda5bb9c1a76a35d7fc6ffc044c07eb206e596)

Reply:
Looks good! This does bring something up, **that we could address at a later time and not blocking for this PR**, linked submissions that changed from `CONTRACT_AND_RATES` to `CONTRACT_ONLY`. Would it even be a common occurrence? How or should we handle that?